src/memjs/memjs.ts

```diff
@@ -0,0 +1,1777 @@
+  //
+  // The callback signature always follows:
+  //
+  //     callback(err, [arg1[, arg2[, arg3[...]]]])
+  //
+  // In case of an error the _err_ argument will be non-null and contain the
+  // `Error`. A notable exception includes a `Client#get` on a key that doesn't
+  // exist. In this case, _err_ will be null, as will the _value and _extras_
+  // arguments.
+
+  /**
+   * GET
+   *
+   * Retrieves the value at the given key in memcache.
+   *
+   * The callback signature is:
+   *
+   *     callback(err, value, flags)
+   *
+   * _value_ and _flags_ are both `Buffer`s. If the key is not found, the
+   * callback is invoked with null for both arguments and no error
+   * @param key
+   * @param callback
+   */
+  get(key: string): Promise<{ value: Value | null; flags: Extras | null }>;
+  get(
+    key: string,
+    callback: (
+      error: Error | null,
+      value: Value | null,
+      flags: Extras | null
+    ) => void
+  ): void;
+  get(
+    key: string,
+    callback?: (
+      error: Error | null,
+      value: Value | null,
+      flags: Extras | null
+    ) => void
+  ): Promise<{ value: Value | null; flags: Extras | null }> | void {
+    var self = this;
+    if (callback === undefined) {
+      return promisify(function (callback) {
+        self.get(key, function (err, value, flags) {
+          callback(err, { value: value, flags: flags });
+        });
+      });
+    }
+    var logger = this.options.logger;
+    this.incrSeq();
+    var request = makeRequestBuffer(constants.OP_GET, key, "", "", this.seq);
+    this.perform(key, request, this.seq, function (err, response) {
+      if (err) {
+        if (callback) {
+          callback(err, null, null);
+        }
+        return;
+      }
+      switch (response!.header.status) {
+        case constants.RESPONSE_STATUS_SUCCCESS:
+          if (callback) {
+            var deserialized = self.serializer.deserialize(
+              response!.header.opcode,
+              response!.val,
+              response!.extras
+            );
+            callback(null, deserialized.value, deserialized.extras);
+          }
+          break;
+        case constants.RESPONSE_STATUS_KEY_NOT_FOUND:
+          if (callback) {
+            callback(null, null, null);
+          }
+          break;
+        default:
+          var errorMessage =
+            "MemJS GET: " + errors[response!.header.status || UNKNOWN_ERROR];
+          logger.log(errorMessage);
+          if (callback) {
+            callback(new Error(errorMessage), null, null);
+          }
+      }
+    });
+  }
+
+  /** Build a pipelined get multi request by sending one GETKQ for each key (quiet, meaning it won't respond if the value is missing) followed by a no-op to force a response (and to give us a sentinel response that the pipeline is done)
+   *
+   * cf https://github.com/couchbase/memcached/blob/master/docs/BinaryProtocol.md#0x0d-getkq-get-with-key-quietly
+   */
+  _buildGetMultiRequest(keys: string[]) {
```

Comment:
```suggestion
  _buildGetMultiRequest(keys: string[]): Buffer {
```

Reply:
sorry I really like explicit return types to ensure readability and lack of bizarre bugs in the future

Reply:
there is an eslint rule for this we could turn on